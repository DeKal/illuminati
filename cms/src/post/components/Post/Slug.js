import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Slug = ({ isEdited, slug, editPostSlug }) => (
  <TextField
    label="Slug"
    required
    variant="outlined"
    value={slug}
    fullWidth
    error={isEdited}
    onChange={event => editPostSlug(event.target.value)}
  />
)

export default Slug

Slug.propTypes = {
  isEdited: PropTypes.bool,
  slug: PropTypes.string,
  editPostSlug: PropTypes.func
}
