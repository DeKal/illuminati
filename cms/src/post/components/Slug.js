import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/base/TextField'

const Slug = ({ isEdited, slug, editPostSlug }) => (
  <TextField
    label="Slug"
    variant="outlined"
    value={slug}
    fullWidth
    error={isEdited}
    onChange={event => editPostSlug(event.target.value)}
  />
)

export default Slug

Slug.propTypes = {
  isEdited: PropTypes.bool,
  slug: PropTypes.string,
  editPostSlug: PropTypes.func
}
