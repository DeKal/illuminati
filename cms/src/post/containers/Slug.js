import { connect } from 'react-redux'
import { getPostSlug, getEditedSlug } from 'post/state/selectors'
import { editPostSlug } from 'post/state/actions'
import Slug from 'post/components/Slug'

const mapStateToProps = state => {
  const editedSlug = getEditedSlug(state)
  const slug = getPostSlug(state)
  return {
    slug: editedSlug || '',
    isEdited: editedSlug !== slug
  }
}

const mapDispatchToProps = {
  editPostSlug
}

export default connect(mapStateToProps, mapDispatchToProps)(Slug)
