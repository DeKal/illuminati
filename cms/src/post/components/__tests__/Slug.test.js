import * as React from 'react'
import { shallow } from 'enzyme'
import Slug from '../Slug'

describe('Slug', () => {
  it('should render Slug correctly', function() {
    const wrap = shallow(<Slug slug="slug" />)
    expect(wrap).toMatchSnapshot()
  })
})
