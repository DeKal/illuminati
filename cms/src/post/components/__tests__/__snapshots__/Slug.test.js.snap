// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slug should render Slug correctly 1`] = `
<WithStyles(WithStyles(ForwardRef(TextField)))
  fullWidth={true}
  label="Slug"
  onChange={[Function]}
  value="slug"
  variant="outlined"
/>
`;
